src/tqdm_publisher/_demo/_server.py

```diff
@@ -0,0 +1,89 @@
+import asyncio
+import json
+import threading
+import time
+from typing import Any, Dict, List
+from uuid import uuid4
+
+import websockets
+
+from tqdm_publisher import TQDMPublisher
+
+
+def start_progress_bar(*, client_id: str, progress_bar_id: str, client_callback: callable) -> None:
+    """
+    Emulate running the specified number of tasks by sleeping the specified amount of time on each iteration.
+
+    Defaults are chosen for a deterministic and regular update period of one second for a total time of one minute.
+    """
+    all_task_durations_in_seconds = [1.0 for _ in range(60)]  # One minute at one second per update
+    progress_bar = TQDMPublisher(iterable=all_task_durations_in_seconds)
+
+    def run_function_on_progress_update(format_dict: dict) -> None:
+        """
+        This is the injected callback that will be run on each update of the TQDM object.
+
+        Its first and only positional argument must be the `format_dict` of the TQDM instance. Additional customization
+        on outside parameters must be achieved by defining those fields at an outer scope and defining this
+        server-specific callback inside the local scope.
+
+        In this demo, we will execute the `client_callback` whose protocol is known only to the WebSocketHandler.
+        It has
+        """
+        client_callback(client_id=client_id, progress_bar_id=progress_bar_id, format_dict=format_dict)
+
+    progress_bar.subscribe(callback=run_function_on_progress_update)
+
+    for task_duration in progress_bar:
+        time.sleep(task_duration)
+
+
+class WebSocketHandler:
+    """Describe this class."""
```

Comment:
Also if you could fill in some of these that would be great - I've never worked with this structure (or `websockets` library) before

Reply:
Thanks